otx/algorithms/detection/adapters/mmdet/task.py

```diff
@@ -331,10 +327,10 @@ def _infer_model(
         cfg = self.configure(False, "test", None)
         logger.info("infer!")
 
-        samples_per_gpu = 1
-
         # Data loader
         mm_dataset = build_dataset(cfg.data.test)
+        samples_per_gpu = cfg.data.test_dataloader.get("samples_per_gpu", 1)
+        samples_per_gpu = samples_per_gpu if len(mm_dataset) % samples_per_gpu == 0 else 1
```

Comment:
how about to add a comment the reason why this code is necessary?

Reply:
Agree with this comment.